cmd/bootkube/recover.go

```diff
@@ -66,6 +70,31 @@ func runCmdRecover(cmd *cobra.Command, args []string) error {
 			return err
 		}
 		backend = recovery.NewEtcdBackend(etcdClient, recoverOpts.etcdPrefix)
+	case recoverOpts.etcdBackupFile != "":
+		bootkube.UserOutput("Attempting recovery using etcd backup file at %q...\n", recoverOpts.etcdBackupFile)
+
+		err := recovery.StartRecoveryEtcdForBackup(recoverOpts.podManifestPath, recoverOpts.etcdBackupFile)
+		if err != nil {
+			return err
+		}
+		defer func() {
+			err := recovery.CleanRecoveryEtcd(recoverOpts.podManifestPath)
+			if err != nil {
+				bootkube.UserOutput("Failed to cleanup recovery etcd from the podManifestPath %v\n", recoverOpts.podManifestPath)
+			}
+		}()
+
+		bootkube.UserOutput("Waiting for etcd server to start...\n")
+		// TODO: better way to detect the startup...
```

Comment:
I have a poll loop in pkg/util/etcdutil/migrate.go#createBootstrapEtcdService that we can factor out